### PR TITLE
Ensure that temporary directories are deleted during tests

### PR DESCRIPTION
**Pull request by eolivelli**
### Motivation
The test suite (especially pulsar-broker) creates lots of temporary files, that accumulate on CI and also they fill up the /tmp dir during development (if you dare to run all of the tests locally)

### Modifications

Modify LocalBookKeeperEnsemble to delete all the temporary directories it created at boot.
We are not deleting directories that are passed externally from the constructor (like in Pulsar Standalone)

### Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/AdvertisedAddressTest.java [hunk 1]
```diff
@@ -42,7 +42,7 @@
 
     @BeforeMethod
     public void setup() throws Exception {
-        bkEnsemble = new LocalBookkeeperEnsemble(3, 0, () -> 0);
+        bkEnsemble = new LocalBookkeeperEnsemble(1, 0, () -> 0);
```
Comment (eolivelli) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/AdvertisedAddressTest.java:45]: a change to this file is not necessary in the scope of this PR, btw we can save to start 2 bookies for this little test

**Review by merlimat — APPROVED** [review]

**Review by sijie — APPROVED** [review]

**Pull request merged**